### DISCUSSION TITLE
How to get the Editor to recognize updated external state variables

### DISCUSSION THREAD

**Discussion by jedgrant** [ueberdosis/tiptap#3015] [General Questions]
This is likely an issue with my inexperience using React generally, but I'd still greatly appreciate any insights.

I've added a commenting plugin to the editor. 

1. When I create a new comment, it creates a DB record for the comment and I store that new comment in a state value (React) which is an array of all comments. 
2. Then I return the ID which I use in a setComment (Mark) command that wraps the selection in a span with a `commentId` on the data-comment attribute. 
3. When I click on that span, I can get the ID value, but the editor `selectionUpdate` function doesn't see the updated value. The page can access it fine, but that function can't see it until the page is reloaded. 

**How do I convince the editor to recognize the updated value in that function?**
A demo minimal app can be found here:
https://codesandbox.io/s/tiptap-react-forked-7fir0b?file=/src/App.js

Help?

**Comment by duythinh-dev** [comment]
Have you solved this problem? I am having the same problem as you :(
- https://github.com/ueberdosis/tiptap/discussions/3196
